src/d3a/models/strategy/area_agents/two_sided_pay_as_bid_agent.py

```diff
@@ -25,10 +25,16 @@ class TwoSidedPayAsBidAgent(OneSidedAgent):
 
     def __init__(self, *, owner, higher_market, lower_market,
                  min_offer_age=ConstSettings.IAASettings.MIN_OFFER_AGE,
-                 engine_type=TwoSidedPayAsBidEngine):
-        super().__init__(engine_type=engine_type, owner=owner,
+                 min_bid_age=ConstSettings.IAASettings.MIN_BID_AGE):
+        self.engines = [
+            TwoSidedPayAsBidEngine('High -> Low', higher_market, lower_market, min_offer_age,
+                                   min_bid_age, self),
+            TwoSidedPayAsBidEngine('Low -> High', lower_market, higher_market, min_offer_age,
+                                   min_bid_age, self),
+        ]
+        super().__init__(owner=owner,
                          higher_market=higher_market, lower_market=lower_market,
-                         min_offer_age=min_offer_age)
+                         min_offer_age=min_offer_age, do_create_engine=False)
```

Comment:
I understand your motivation here, but is it maybe possible instead of setting this boolean variable rather send the engine class that the engines should be build of?

Reply:
Or is it something one just does not do, sending a class to the super.__init__() ?

Reply:
The problem is that now the engine classes do not accept the same arguments, therefore there is no way to have a constructor that accepts the same arguments (unless we use the 'creative' and confusing way that the Area/MarketEvents use, by using * arguments). That obliges us to not construct the engine in the base class, but in the respective agent class constructor. The alternative to this boolean flag is to have an optional engine_objects argument, that would pass the engines to the base classes, and if it is empty only then the base class should create its engine. I find it pretty much the same as the approach with the boolean variable, however if you prefer it that way let me know and I'll implement it .

Reply:
OK, yes i get it. No then leave it.
But AFAIS, this boolean you dont need this boolean flag really, as you construct the engines in every agent explicitly anyway. Or did i miss something?

Reply:
The problem is that the agents inherit from one another. Therefore if you construct the two sided market engines in the two sided agent, you need to avoid constructing the one-sided engine from the base class (which is the one-sided agent). That's why sucha  variable is actually required.

Reply:
ah, yes, sorry. You are right of course.